bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -571,6 +583,154 @@ public KNXCoreTypeMapper() {
         defaultDptMap.put(HSBType.class, DPTXlatorRGB.DPT_RGB.getID());
     }
 
+    /*
+     * This function computes the target unit for type conversion from OH quantity type to DPT types.
+     * Calimero library provides units which can be used for most of the DPTs. There are some deviations
+     * from the OH unit scheme which are handled.
+     */
+    private String quantityTypeToDPTValue(QuantityType<?> qt, int mainNumber, int subNumber, String targetOhUnit)
+            throws KNXException {
+        double scaleFactor = 1.0;
+        switch (mainNumber) {
+            case 7:
+                switch (subNumber) {
+                    case 3:
+                    case 4:
+                        targetOhUnit = "ms";
+                        break;
+                }
+                break;
+            case 9:
+                switch (subNumber) {
+                    // special case: temperature deltas specified in different units
+                    // ignore the offset, but run a conversion to handle prefixes like mK
+                    // scaleFactor is needed to properly handle °F
+                    case 2: {
+                        final String unit = qt.getUnit().toString();
+                        // find out if the unit is based on °C or K, getSystemUnit() does not help here as it always
+                        // gives "K"
+                        if (unit.contains("°C")) {
+                            targetOhUnit = "°C";
+                        } else if (unit.contains("°F")) {
+                            targetOhUnit = "°F";
+                            scaleFactor = 5.0 / 9.0;
+                        } else if (unit.contains("K")) {
+                            targetOhUnit = "K";
+                        } else {
+                            targetOhUnit = "";
+                        }
+                    }
+                        break;
```

Comment:
The more common way to do is I believe to put the break inside the block `{}`
Same for other switch cases